### DISCUSSION TITLE
Inline LaTeX

### DISCUSSION THREAD

**Discussion by hjstein** [testingisdocumenting/znai#1222] [Q&A]
The syntax that I see in the documentation for inline LaTeX is something like:
```
Consider the function `:latex: {src: "f(x) = x^2"}`.
```
That's much more cumbersome than
```
Consider the function $f(x) = x^2$.
```

I tried the dollar sign notation & saw that it didn't work.  Is there some other abbreviated notation available for inline LaTeX?

Thanks.

**Comment by MykolaGolubyev** [comment]
I can see the point of shortening the syntax. Will omitting `src` block work?
```
`:latex: f(x) = x^2`
```

Adding completely new syntax would be a bigger lift. Znai uses default markdown and "leverages" existing constructs like "fence block" and "inline code" for plugins.

**Comment by hjstein** [comment]
That helps, but it's not nearly as convenient as $(x) = x^2$ (that is [dollar][expression][dollar], or even [dollar][backquote][expression][dollar][backquote], both of which are often supported in markdown processors.

**Reply by MykolaGolubyev** [reply]
Could you please point me towards a tool that support this syntax? I am using common mark java parsing library for Java from attlasian. Will check if someone added extension for it

**Reply by hjstein** [reply]
What's used for markdown rendering in github & gitlab?  They both support the [dollar][expression][dollar]...

**Reply by hjstein** [reply]
https://docs.gitlab.com/ee/user/markdown.html#math:
```
Math written in LaTeX syntax is rendered with [KaTeX](https://github.com/KaTeX/KaTeX).
```
It accepts [dollar][expression][dollar] as well as [dollar][backquote][expression][backquote][dollar], and you're also using KaTeX.

But I haven't figured out what parser they're using.  With commonmark-java, evidently, it's not entirely straight-forward, given:

https://github.com/commonmark/commonmark-java/issues/208

**Reply by hjstein** [reply]
https://docs.gitlab.com/ee/development/gitlab_flavored_markdown/specification_guide/#parsing-and-rendering:


The backend parser / renderer supports parsing and rendering to static read-only HTML. It is [implemented in Ruby](https://gitlab.com/gitlab-org/gitlab/-/tree/master/lib/banzai). It leverages the [commonmarker](https://github.com/gjtorikian/commonmarker) gem, which is a Ruby wrapper for [libcmark-gfm](https://github.com/github/cmark), GitHub’s fork of the reference parser for CommonMark. libcmark-gfm is an extended version of the C reference implementation of [CommonMark](https://commonmark.org/)

**Reply by MykolaGolubyev** [reply]
Thanks for looking into this! I followed some links and found out that common-mark library added new capabilities to create new inline parsers. Will give it a go.

As an extra context, Znai doesn't generate html right away and doesn't use any global javascript parsing. Instead it creates JSON blocks so the rendering logic can be decided and switched at runtime (i.e. search preview, presentation mode, single page output)

**Comment by hjstein** [comment]
For github, see: https://stackoverflow.com/questions/39560644/what-library-does-github-use-for-parsing-markdown

**Reply by MykolaGolubyev** [reply]
@hjstein I have started a new feature set for upcoming release and working on $ and $$ support. 
Wonder if you are still interested in this

**Reply by MykolaGolubyev** [reply]
https://github.com/testingisdocumenting/znai/pull/1243

**Comment by hjstein** [comment]
I'll try it out.
